cmd/ncproxy/ncproxy.go

```diff
@@ -1,3 +1,5 @@
+//go:build windows
```

Comment:
[Not blocking this pr in anyway] I can only comment on changed lines, but @katiewasnothere it'd be worth splitting the hcn networking from the ncproxynetworking implementations into _windows and _linux files most likely (at least that's what my memory is telling me the github.com/Microsoft/hcsshim/internal/ncproxy/networking definitions were for). 

We also use winio in this file as well for `DialPipe`, which I guess on Linux we could use a Unix domain socket instead.